net/redisclient.go

```diff
@@ -87,6 +101,96 @@ const (
 	defaultConnMetricsInterval = 60 * time.Second
 )
 
+// JumpHash
+// https://arxiv.org/pdf/1406.2294.pdf
+type JumpHash struct {
+	hash   hash.Hash64
+	shards []string
+}
+
+func NewJumpHash(shards []string) redis.ConsistentHash {
+	return &JumpHash{
+		hash:   fnv.New64(),
+		shards: shards,
+	}
+}
+
+func (j *JumpHash) Get(k string) string {
+	_, err := j.hash.Write([]byte(k))
+	if err != nil {
+		log.Fatalf("Failed to write %s to hash: %v", k, err)
+	}
+
+	key := j.hash.Sum64()
+	h := jump.Hash(key, len(j.shards))
+	return j.shards[int(h)%len(j.shards)]
+}
+
+// Multi-probe consistent hashing - mpchash
+// https://arxiv.org/pdf/1505.00062.pdf
+type multiprobe struct {
+	hash   *mpchash.Multi
+	shards []string
+}
+
+func NewMultiprobe(shards []string) redis.ConsistentHash {
+	return &multiprobe{
+		// 2 seeds and k=21
```

Comment:
21 seems arbitrary, why not k=100?
Also
> We propose to store each node once but to hash keys K ways, returning the subsequent
node which is closest to a key hash. We call this multi-probe consistent hashing. This
requires O(n) space to store n nodes and O(K) time per lookup. Perhaps surprisingly,
using only K = 2 key hashes improves the peak-to-average load ratio from O(ln n) to O(1)
with high probability.

Reply:
It's 21 in the library itself in their tests/benchmarks, so I stick with that

Reply:
made a comment